src/signTx.c

```diff
@@ -234,6 +235,15 @@ static inline void advanceStage()
 			// we are not waiting for any APDU here, network id is already known from the init APDU
 			txHashBuilder_addNetworkId(&BODY_CTX->txHashBuilder, ctx->commonTxData.networkId);
 		}
+		ctx->stage = SIGN_STAGE_BODY_TOTAL_COLLATERALS;
+		break;
+
+	case SIGN_STAGE_BODY_TOTAL_COLLATERALS:
+		if (ctx->includeTotalCollateral) {
+			// TODO: Any assertion here? (sum of ADA in collateral inputs) = txColl + (ADA in collRet)
+			txHashBuilder_enterTotalCollaterals(&BODY_CTX->txHashBuilder, ctx->txColl);
+		}
+		//TODO: should we assert for (sum of ADA in collateral inputs) >= (min collateral requirement) + (ADA in collRet) as the "else" case?
```

Comment:
Does not belong in here, see `securityPolicy.c` where such conditions are checked and then applied (resulting in UI warnig messages, tx rejection etc.)